packages/drivers/odsp-driver/src/graph.ts

```diff
@@ -88,66 +150,37 @@ export async function getShareLink(
  * @param nameForLogging - Name used for logging
  * @param logger - used to log results of operation, including any error
  * @param requestInit - Request Init to be passed to fetch
- * @param retryPolicy - Retry policy to be passed to fetchWithRetry
- * @param timeoutMs - Timeout value to be passed to fetchWithRetry
  */
 export async function graphFetch(
     getShareLinkToken:
         (options: TokenFetchOptions, scopeFor: SharingLinkScopeFor, siteUrl: string) => Promise<string | null>,
     siteUrl: string,
     graphUrl: string,
     nameForLogging: string,
-    logger?: ITelemetryLogger,
+    logger: ITelemetryLogger,
     requestInit?: RequestInit,
-    retryPolicy?: RetryPolicy<Response>,
-    timeoutMs = 0,
-): Promise<Response> {
-    const getToken = async (options: TokenFetchOptions) =>
-        getShareLinkToken(
-            options,
-            SharingLinkScopeFor.nonFileDefaultUrl,
-            siteUrl,
+): Promise<IOdspResponse<IGraphFetchResponse>> {
+    const response = await getWithRetryForTokenRefresh(async (options) => {
+        const odspResponse = await PerformanceEvent.timedExecAsync(logger,
+            { eventName: "odspFetchResponse", requestName: nameForLogging }, async (event) => {
+                const startTime = performance.now();
+                const token = await getShareLinkToken(options, SharingLinkScopeFor.nonFileDefaultUrl, siteUrl);
+                const { url, headers } = getUrlAndHeadersWithAuth(graphUrl.startsWith("http")
+                    ? graphUrl : `https://graph.microsoft.com/v1.0/${graphUrl}`, token);
+                const augmentedRequest = { ...requestInit };
+                augmentedRequest.headers = { ...augmentedRequest.headers, ...headers };
+                const res = await fetchAndParseAsJSONHelper<IGraphFetchResponse>(url, augmentedRequest);
+                const totalTime = performance.now() - startTime;
```

Comment:
You should not collect time measure that way, PerformanceEvent will do it for you.
Also in places where we need to do something like that, please always use "duration" property - not only it's easier to find in telemetry, but our logger will format it properly.

Reply:
As I checked in odsp storage service, we put Performance event inside getWithRetryForTokenRefresh. So if the call throws, the performance event will be logged as canceled and on success as end. So we could still count successful calls and canceled events per session?

Reply:
I guess here we can try other way as you said and log no retries it took by logging it in count and reduce telemetry events.